### DISCUSSION TITLE
memcpy_impl.S: No such file or directory

### DISCUSSION THREAD

**Discussion by JunJianX** [Motion-Project/motion#1782] [Q&A] [answered]
### Description

[Provide a clear and concise description of the problem. Include any relevant details about the circumstances under which the issue occurred.]

### Steps to Reproduce

1. [First step]
2. [Second step]
3. [And so on...]

### Expected Behavior

[Describe what you expected to happen.]

### Actual Behavior
I have installed motion by `sudo apt-get install motion`,which is normal.
But When I compile motion myself,I encounter a question below.
Here are the debug backtrace.

`Thread 4 "ml1" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xaca8bca0 (LWP 7124)]
0xb59978ec in __memcpy_neon () at ../sysdeps/arm/armv7/multiarch/memcpy_impl.S:492
492     ../sysdeps/arm/armv7/multiarch/memcpy_impl.S: No such file or directory.
(gdb) bt
#0  0xb59978ec in __memcpy_neon () at ../sysdeps/arm/armv7/multiarch/memcpy_impl.S:492

#1  0x00424f12 in memcpy (__len=<optimized out>, __src=<optimized out>, __dest=0xabc99008) at /usr/include/arm-linux-gnueabihf/bits/string_fortified.h:34

#2  v4l2_pix_change (
    dest=0xabc99008 '\004' <repeats 19 times>, "\002\003\003\002\002\004\004\004\004\004\004\004\004\003\003\004\003\003\004\003\003\003", '\004' <repeats 39 times>, "\002\002\004\002\002\003\004\002\002\003\002\002\003\003", '\004' <repeats 21 times>, "\003\003\003\004\002\003\003\002\002", '\004' <repeats 15 times>, "\003\003\004\003\003", '\004' <repeats 35 times>, "\003\003\003\003\003\003\003\003\003\004\004\004\004\004\004\003\003\004\003\003\004"..., curdev=0xac100638, cnt=0x4778a0) at video_v4l2.c:1209

#3  v4l2_next (cnt=0x4778a0, img_data=0x47ab58) at video_v4l2.c:1809
#4  0x004107da in motion_init (cnt=0x4778a0) at motion.c:1219

#5  motion_loop (arg=0x4778a0) at motion.c:2791

#6  0xb5a3b9d8 in start_thread (arg=0x134307d8) at pthread_create.c:477

#7  0xb59d494c in ?? () at ../sysdeps/unix/sysv/linux/arm/clone.S:73 from /lib/arm-linux-gnueabihf/libc.so.6
Backtrace stopped: previous frame identical to this frame (corrupt stack?)`

It seems like a memcpy error.But Why?
### Versions

- **Software Version**: 629b3babf0d0375592de61f2f05c85b460efd65c
- **Operating System**: Armbian
- **Hardware Platform**: Linux orangepipc 5.4.65-sunxi #2.2.2 SMP Fri Jun 7 17:40:23 CST 2024 armv7l armv7l armv7l GNU/Linux

**Comment by tosiara** [comment]
FWIW, I run self-compiled motion on OrangePi Armbian with USB camera, no issues.
Try to change camera to see if there is any effect.
Also, please attach full compile logs and motion log with -d 9

**Reply by JunJianX** [reply]
The difference might be I use a CSI camera ,type ov2640.

I moved compiled file motion to a dir,with conf file.
`orangepi@orangepipc:~/code/motion/test$ ls                                                         
motion  motion.conf  motion.log`
and run with `sudo ./motion -c motion.conf -d 9` 
Here are the log and conf file(with a appendix '.log' ,cause github don't support conf format upload.)
[motion.log](https://github.com/user-attachments/files/16138085/motion.log)
[motion.conf.log](https://github.com/user-attachments/files/16138088/motion.conf.log)
and this is what I compiled ,with debug info
`motion: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, BuildID[sha1]=d7d46fead3aa6bda01d1b880a6dd413af5439072, for GNU/Linux 3.2.0, with debug_info, not stripped`
[https://drive.google.com/file/d/1oQtG3_1RLJJroKvcskeHh0AmiH_5hZ3Y/view?usp=drive_link](url)

**Reply by JunJianX** [reply]
@tosiara Any ideas?

**Comment by JunJianX** [comment, marked as answer]
I kwon the reason,cause I did not use a release version
